### ISSUE TITLE
carla - segmentation fault

### ISSUE THREAD

**Issue by pinootto** [falkTX/Carla#1405]
Hello,

I am on slackware64 stable 14.2 and I am trying to install carla from the sbo slackbuild.

I tried with the version 2.1 of carla in https://slackbuilds.org/repository/1.../?search=carla and also with the new version 2.3.0 (downloaded the source code and compiled successfully), but I always get a segmentation fault:

giovanni@slackserver:~$ carla
LRDF Support not available (LADSPA-RDF will be disabled)
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-giovanni'
Carla 2.3.0 started, status:
Python version: 3.7.2
**Qt version: 5.14.0
PyQt version: 5.14.1**
Binary dir: /usr/lib64/carla
Resources dir: /usr/share/carla/resources
Frontend pixel ratio is 1.0
libjack.so.0 loaded successfully!
Segmentation fault

**The strange thing is that carla says Qt version: 5.14.0 and PyQt version: 5.14.1, but actually I don't have those versions installed.**

My installed versions are:

root@slackserver:~# slpkg -f qt5

Packages with matching name [ qt5 ]

[ installed ] [ 15M ] - qbittorrent-qt5-4.2.5-x86_64-1_SBo
[ installed ] [ 341M ] - **qt5-5.12.8-x86_64-1_SBo**
[ installed ] [ 37M ] - qt5-webkit-5.9.1-x86_64-1alien
[ installed ] [ 770K ] - quazip-qt5-0.8.1-x86_64-1_SBo

and

root@slackserver:~# slpkg -f PyQt

Packages with matching name [ PyQt ]

[ installed ] [ 40M ] - PyQt-4.11.4-x86_64-1
[ installed ] [ 29M ] - **python3-PyQt5-5.13.2-x86_64-2_SBo**

Do you have any idea/suggestion about how to fix this problem?

Where carla is searching for the Qt and PyQt versions?

I really need to use carla for LMMS.

Best regards,
Giovanni

**Comment by falkTX** [comment]
please build carla in debug mode.
```
make clean
make debug
# and install as usual
```
then you can use it inside gdb to get a backtrace
```
carla --gdb
# run, bt
```

**Comment by pinootto** [comment]
Good news!

I have succeeded to make carla work!

In the python home directory I deleted some sub-directories related to previous (dirty) installations of:
   python3-sip
   python3-PyQt5

then I recompiled them to:
   python3-sip-4.19.20-x86_64-3_SBo
   python3-PyQt5-5.13.2-x86_64-2_SBo

then I recompiled carla to:
   carla-2.1-x86_64-1_SBo

So now carla is executing correctly and showing the right versions of Qt and PyQt:

giovanni@slackserver:~$ carla
LRDF Support not available (LADSPA-RDF will be disabled)
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-giovanni'
Carla 2.1.0 started, status:
  Python version: 3.7.2
  Qt version:     5.12.8
  PyQt version:   5.13.2
  Binary dir:     /usr/lib64/carla
  Resources dir:  /usr/share/carla/resources
qt.svg: <input>:552:178: Could not resolve property: #linearGradient5167
qt.svg: <input>:30274: Could not resolve property: #linearGradient5167
qt.svg: <input>:30274: Could not resolve property: #pattern5614
qt.svg: <input>:30274: Could not resolve property: #pattern5626
qt.svg: <input>:5672: Could not resolve property: #radialGradient3709
qt.svg: <input>:3226: Could not resolve property: #linearGradient3893-1
qt.svg: <input>:3190: Could not resolve property: #linearGradient3893-1
qt.svg: <input>:3208: Could not resolve property: #linearGradient3893-1
Frontend pixel ratio is 1.0
libjack.so.0 loaded successfully!
No device set, using 2 inputs and 2 outputs

I don't know why I had those strange installations of python3-sip and python3-PyQt5.

Anyway, now the system seems cleaner.

Thank you for your support.
